src/functions/files.ts

```diff
@@ -280,6 +314,128 @@ export const reUploadImageToS3AndRemove = async (startAt: number = 0) => {
   }
 }
 
+/**
+ * flow this document https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#listObjects-property
+ * @param limit Sets the maximum number of keys returned in the response. The response might contain fewer keys but will never contain more.
+ * @param marker Specifies the key to start with when listing objects in a bucket.
+ */
+export const listObjects = ({
+  limit,
+  marker,
+}: {
+  limit?: number
+  marker?: string
+} = {}) => {
+  return s3
+    .listObjects({
+      Bucket,
+      // up to 1000
+      MaxKeys: limit,
+      Marker: marker,
+    })
+    .promise()
+}
+
+export const listAllObject = async () => {
+  let marker
+  let isTruncated = true
+  let objects: S3.Object[] = []
+  let count = 0
+
+  while (isTruncated) {
+    logger.verbose(
+      `[Get objects from ${count}000-${count + 1}000]: marker ${marker}`,
+    )
+    const response: S3.ListObjectsOutput = await listObjects({
+      marker,
+    })
+
+    objects = objects.concat(response.Contents)
+
+    isTruncated = response.IsTruncated
+
+    if (isTruncated) {
+      marker = response.Contents.slice(-1)[0].Key
+    }
+    count += 1
+  }
+
+  return objects
+}
+
+export const putObjectACL = ({ key, acl }: { key: string; acl: ACL }) => {
+  return s3
+    .putObjectAcl({
+      Bucket,
+      Key: key,
+      ACL: acl,
+    })
+    .promise()
+}
+
+export const rePutAllObjectACL = async (
+  startAt: number = 0,
+  fromMarker?: string,
+) => {
+  let marker = fromMarker
+  let isTruncated = true
+  let count = startAt
+  let errorObjects: S3.Object[] = []
+
+  while (isTruncated) {
```

Comment:
why don't reuse the `listAllObjects` function?

Reply:
because the max limit for each request to get objects be set to 1000 by AWS, follow this document: [listObjects](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#listObjects-property)

we don't need to wait for all objects returned and put ACL, we can put for each 1000 objects